src/reason_parser.mly

```diff
@@ -1617,8 +1617,8 @@ nonlocal_module_binding_details:
 non_arrowed_module_type: mark_position_mty(_non_arrowed_module_type) {$1}
 _non_arrowed_module_type:
   | simple_module_type {$1}
-  | MODULE TYPE module_expr %prec below_LBRACKETAT
-      { mkmty(Pmty_typeof $3) }
+  | MODULE TYPE OF module_expr %prec below_LBRACKETAT
```

Comment:
I'm not sure but I doubt it because even if converting from `.ml`, the printer printed it as `module type of`, which wouldn't have parsed. Likewise, even if someone wrote it in `.re`, the printer would be printing it in a form that wouldn't have parsed.